common/src/api/external/mod.rs

```diff
@@ -1819,11 +1831,14 @@ impl JsonSchema for MacAddr {
 }
 
 /// A `NetworkInterface` represents a virtual network interface device.
```

Comment:
As implemented we hang on to these with static IPs for the duration of the instance's lifetime. I don't know if this is desirable behavior for longer term, but without it we'll need to have a plan for handling boot-time IP allocation failures.

We'd likely want to keep the rows live in the table even if there isn't always an IP associated with them, since VMs expect to have the same set of interfaces to reappear on each boot (including likely with the same MACs)

Reply:
So, a network interface that are using IP addresses from a VPC Subnet (that is the IP address that we see internally in instances in ip addr, ifconfig, etc.) are associated and allocated on instance creation and freed on instance destruction. External IPs (which will may to the primary aka first of these) are what change at boot time unless a floating ip is used.